### ISSUE TITLE
Update exact synonyms QC

### ISSUE THREAD

**Issue by sabrinatoro** [monarch-initiative/mondo#6444]
The QC checks to identify duplicate exact synonyms should be updated to 
- be case insensitive
- not take dashes and spaces in the synonym names into account
- ignore word order (though this might be out of scope)

Examples of current terms with same/almost exact synonyms 
- ('MONDO:0007964', 'EXACT', 'B-K Mole syndrome'), ('MONDO:0018453', 'EXACT', 'B-K mole syndrome')
- ('MONDO:0008408', 'EXACT', 'scapuloperoneal spinal muscular atrophy'), ('MONDO:0010058', 'EXACT', 'spinal muscular atrophy, scapuloperoneal')

**Comment by matentzn** [comment]
@hrshdhgd I optimistically thoought I could do this with SPARQL but this is not possible. As we will need more extensive QC moving forward, can you implement this here in https://github.com/monarch-initiative/mondolib (`mondo qc --exact-synonyms`, or some such)? And add a make goal here that to run the test?

**Comment by matentzn** [comment]
As per questions on slack:

the input is an ontology (mondo.db, based of mondo.owl), and the output is `logging.warnings()` according the the above rules. As a first pass: Identify synonyms that are the same after the same basic processing you do in synonymiser, e.g. casing, whitespace standardisation, etc. You should be able to identify cases such as

`B-K Mole syndrome`, `B-K mole syndrome` and `B-K mole      syndrome`

**Comment by cmungall** [comment]
Lexical variants: it would be good to have more OMO guidance here. But I like having common lexical variants, even when trivial, it makes terms more findable when using a broad range of tools. However, we could potentially tag these. I think also permitted/forbidden lexical variants should very much be on a per-DP basis.

Having said that, I am not a fan of synonyms like "spinal muscular atrophy, scapuloperoneal" regardless of whether they are duplicates or not. These are typically legacy terms introduced to facilitate maintaining groupings on an alphabetic sort. 

Whitespace standardization: the existing OAK `validate` command should handle cases such as double whitespace.

however we should not have two synonyms that only vary in capitalization.

I would rather standardize as much of this in OAK as possible no need for mondo to have bespoke rules.

**Comment by cmungall** [comment]
chatted with @hrshdhgd as a first pass we will use lexmatch to do mondo-vs-mondo, this can give us a sense of which terms match up in case/lexical-order independent ways

**Comment by hrshdhgd** [comment]
I've run `runoak validate` and `runoak lexmatch -R rules-file-from-mondo-ingest` on `mondo-edit.obo` 

- [validate-result](https://github.com/monarch-initiative/mondolib/blob/main/tmp/mondo_validate.tsv)
- [lexmatch result](https://docs.google.com/spreadsheets/d/19ftL9oX4RkuF3qbHtBdr1EyB-gg30jJo4gLAseaEAo0/edit?usp=sharing)

**Comment by matentzn** [comment]
Absolutely fantastic @hrshdhgd.

I made this follow-up issue: 

https://github.com/monarch-initiative/mondo/issues/5994#issuecomment-1689689482

I would say, once this is fixed, we come back to this issue here and think about implementing QC.

https://github.com/monarch-initiative/mondolib/blob/main/tmp/mondo_validate.tsv is very interesting.

Since you already did start working on it, lets push the envelop:

1. Remove cases where entities are included that are not MONDO ids
2. Incorrect object type for has_obsolescence_reason range = string should_be_iri = False: wrong, should be IRI
3. Incorrect datatype for term_tracker_item expected: xsd:string for string: WRONG should be IRI
4. Incorrect object type for disjointWith range = string should_be_iri = False: WHAT does this mean?

Let's start with these; Thank you!

**Comment by hrshdhgd** [comment]
1. Completed bullet 1.
  i. I re-ran the whole pipeline and all rows with `subject` with `MONDO:xxxxx` disappeared. Is this expected?
2. Regarding everything else, I don't understand what are the action items.

**Comment by matentzn** [comment]
Maybe 1 is done in reverse? If you removed everything other than Mondo IDs the new results should _only_ have rows with Mondo IDs!

**Comment by matentzn** [comment]
The rest is no rush, can you mention in next call?

**Comment by hrshdhgd** [comment]
> Maybe 1 is done in reverse? If you removed everything other than Mondo IDs the new results should _only_ have rows with Mondo IDs!

Nope. The new result is blank (https://github.com/monarch-initiative/mondolib/blob/main/tmp/mondo_validate.tsv)

> The rest is no rush, can you mention in next call?

Sure

**Comment by matentzn** [comment]
It's hard to imagine that one day to the next all these Mondo errors that were previously there are now fixed! You could check the ontology and see if an error from a previous run is still present _and_ does not show up anymore now?

**Comment by hrshdhgd** [comment]
Fixed! Somehow I ended up with very old versions of `oaklib` and `semsql` which had bugs in the past. It is all sorted now and you you can see the updated files:

1. [Only MONDO](https://github.com/monarch-initiative/mondolib/blob/main/tmp/mondo_validate.tsv)
2. [All](https://github.com/monarch-initiative/mondolib/blob/main/tmp/mondo_validate_full.tsv)

**Comment by cmungall** [comment]
type|id|severity|property|-|-|-|message
---|---|---|---|---|---|---|---
http://www.w3.org/ns/shacl#DatatypeConstraintComponent | MONDO:0000218 | ERROR |   | IAO:0000231 |   |   |   | Incorrect object type for has_obsolescence_reason range = string should_be_iri = False

we'll make the reporting clearer. This is not a false positive. At the time this was ran the range of IAO:0000231  for MONDO:0000218 was a URI not a string, as expected.

I see this is still true for other terms

id: MONDO:0000991
name: obsolete left bundle branch block
def: "OBSOLETE. A bundle branch block in which the activation of the left ventricle is delayed." [MONDO:cjm, Wikipedia:Left_bundle_branch_block]
synonym: "left bundle branch [block] or [hemiblock]" RELATED [DOID:10272]
synonym: "left bundle branch hemiblock" RELATED [DOID:10272]
xref: DOID:10272 {source="MONDO:obsoleteEquivalent"}
xref: ICD10CM:I44.60 {source="DOID:10272"}
xref: ICD9:426.2 {source="MONDO:obsoleteEquivalent", source="MONDO:i2s", source="DOID:10272"}
xref: SCTID:195044001 {source="DOID:10272"}
xref: SCTID:195045000 {source="DOID:10272"}
xref: SCTID:266245009 {source="DOID:10272"}
xref: SCTID:4973001 {source="MONDO:obsoleteEquivalent", source="DOID:10272"}
xref: SCTID:63467002 {source="MONDO:obsoleteEquivalent"}
xref: UMLS:C0155702 {source="MONDO:obsoleteEquivalent", source="DOID:10272"}
property_value: IAO:0000231 IAO:0000423
property_value: IAO:0000233 "https://github.com/monarch-initiative/mondo/issues/2536" xsd:anyURI
is_obsolete: true
consider: HP:0011713

if this is not desired behavior, change the omo schema in oak

**Comment by matentzn** [comment]
Very nice, we will discuss this at the next tech call!
